api/scpca_portal/models/library.py

```diff
@@ -183,3 +183,16 @@ def get_combined_library_metadata(self) -> List[Dict]:
             self.project.get_metadata() | sample.get_metadata() | self.get_metadata()
             for sample in self.samples
         ]
+
+    def get_filtered_data_file_paths(self, download_config: Dict) -> List[Path]:
+        file_extension_to_be_excluded = (
+            common.ANNDATA_EXT
+            if download_config["format"] == Library.FileFormats.SINGLE_CELL_EXPERIMENT
+            else common.SCE_EXT
+        )
+
+        return [
+            file_path
+            for file_path in [Path(fp) for fp in self.data_file_paths]
+            if file_path.suffix != file_extension_to_be_excluded
+        ]
```

Comment:
Just a few suggestions for this:

- rename to `get_download_config_file_paths` so we can specify why subset of filepaths are being returned
- if we keep this implementation I would recommend using `omit_suffix` instead of `file_extension_to_be_excluded`

Maybe we should consider putting the Format names and extensions relationship in `common.py`.

```python
# common.py
# Formats
ANN_DATA = "ANN_DATA"
SINGLE_CELL_EXPERIMENT = "SINGLE_CELL_EXPERIMENT"

FORMAT_EXTENSIONS = {
    ANN_DATA: ".h5ad",
    SINGLE_CELL_EXPERIMENT: ".rds"
}
```

Then we could do something like the following which would allow us to avoid the "if this then the other one" flavor 
of logic in favor of using sets.

```python

    def get_filtered_data_file_paths(self, download_config: Dict) -> List[Path]:
    
        omit_suffixes = set(common.FORMAT_EXTENSIONS.values())
        omit_suffixes.remove(common.FORMAT_EXTENSIONS[download_config["format"]])
        
        return [
            file_path
            for file_path in [Path(fp) for fp in self.data_file_paths]
            if file_path.suffix not in omit_suffixes
        ]
```

Additionally, I don't think that any of the models should "own" the definition of the `ANN_DATA` constant and I'd prefer not to have it defined in more than one place if possible. This does seem to open up more discussion than we strictly need to address here but I think we should consider it before we use `common.ANNDATA_EXT` more broadly.

Reply:
I love the idea of moving the format constants `ANN_DATA` and `SINGLE_CELL_EXPERIMENT` outside of the models and into `common.py`. What do you think about moving the Modality constants there also?

For context, here are the places where constants are defined. Doesn't this seem redundant while also introducing unnecessary dependencies?

`computed_file.py`
```python
class OutputFileModalities:
    SINGLE_CELL = "SINGLE_CELL"
    SPATIAL = "SPATIAL"

    CHOICES = (
        (SINGLE_CELL, "Single Cell"),
        (SPATIAL, "Spatial"),
    )

class OutputFileFormats:
    ANN_DATA = "ANN_DATA"
    SINGLE_CELL_EXPERIMENT = "SINGLE_CELL_EXPERIMENT"

    CHOICES = (
        (ANN_DATA, "AnnData"),
        (SINGLE_CELL_EXPERIMENT, "Single cell experiment"),
    )
```

`sample.py`
```python
class Modalities:
    BULK_RNA_SEQ = "BULK_RNA_SEQ"
    CITE_SEQ = "CITE_SEQ"
    MULTIPLEXED = "MULTIPLEXED"
    SINGLE_CELL = "SINGLE_CELL"
    SPATIAL = "SPATIAL"

    NAME_MAPPING = {
        BULK_RNA_SEQ: "Bulk RNA-seq",
        CITE_SEQ: "CITE-seq",
        MULTIPLEXED: "Multiplexed",
        SPATIAL: "Spatial Data",
    }
```

`library.py`
```python
class FileFormats:
    ANN_DATA = "ANN_DATA"
    SINGLE_CELL_EXPERIMENT = "SINGLE_CELL_EXPERIMENT"

    CHOICES = (
        (ANN_DATA, "AnnData"),
        (SINGLE_CELL_EXPERIMENT, "Single cell experiment"),
    )

class Modalities:
    SINGLE_CELL = "SINGLE_CELL"
    SPATIAL = "SPATIAL"

    CHOICES = (
        (SINGLE_CELL, "Single Cell"),
        (SPATIAL, "Spatial"),
    )

```